src/uucore/src/lib/features/checksum.rs

```diff
@@ -488,8 +514,32 @@ where
             }
         };
 
-        let reader = BufReader::new(file);
-        let lines: Vec<String> = reader.lines().collect::<Result<_, _>>()?;
+        let mut reader = BufReader::new(file);
+        let lines = iter::from_fn(|| {
```

Comment:
maybe move this into a function?

Reply:
I've done it in #6654, if you're OK with letting it like this here I'll be happy, cause I'd prefer to avoid messy rebasing is possible.

Reply:
I moved the change into this PR